src/plugin/settings/engine.cpp

```diff
@@ -204,242 +206,243 @@ Result<const ISettingsResponse *> Engine::settingsReceived()
-            {
-                Engine::pushPluginDiagnosticEvent(IPluginDiagnosticEvent::Level::error, "Plugin Secret Error",
-                                                  "Secret key provided is incorrect: " +
-                                                      parseCloudfuseError(dryRunRet.output));
-                NX_PRINT << "Error with provided endpoint: " + parseCloudfuseError(dryRunRet.output);
-                return error(ErrorCode::otherError,
-                             "Secret key provided is incorrect: " + parseCloudfuseError(dryRunRet.output));
-            }
-
-            // Otherwise this is an error we did not prepare for
-            Engine::pushPluginDiagnosticEvent(IPluginDiagnosticEvent::Level::error, "Plugin Error",
-                                              "Unable to validate credentials with error: " +
-                                                  parseCloudfuseError(dryRunRet.output));
-            NX_PRINT << "Unable to validate credentials with error: " + parseCloudfuseError(dryRunRet.output);
-            return error(ErrorCode::otherError,
-                         "Unable to validate credentials with error: " + parseCloudfuseError(dryRunRet.output));
         }
-        NX_PRINT << "spawning process from mount";
+        // Otherwise this is an error we did not prepare for
+        return error(ErrorCode::internalError,
+                     "Unable to validate credentials with error: " + parseCloudfuseError(dryRunRet.output));
+    }
+    return Error(ErrorCode::noError, nullptr);
+}
+
+nx::sdk::Error Engine::spawnMount()
+{
+    // mount the bucket
+    NX_PRINT << "Starting cloud storage mount";
 #if defined(__linux__)
-        const processReturn mountRet = cfManager.mount(keyId, secretKey, passphrase);
+    const processReturn mountRet = m_cfManager.mount(m_keyId, m_secretKey, m_passphrase);
 #elif defined(_WIN32)
-        const processReturn mountRet = cfManager.mount(passphrase);
+    const processReturn mountRet = m_cfManager.mount(m_passphrase);
 #endif
+    if (mountRet.errCode != 0)
+    {
+        std::string errorMessage = "Unable to launch mount with error: " + mountRet.output;
+        NX_PRINT << errorMessage;
+        return error(ErrorCode::internalError, errorMessage);
+    }
 
-        if (mountRet.errCode != 0)
-        {
-            NX_PRINT << "Unable to launch mount with error: " + mountRet.output;
-            return error(ErrorCode::internalError, "Unable to launch mount with error: " + mountRet.output);
-        }
-
-        // Mount might not show up immediately, so wait for mount to appear
-        int retryCount = 0;
-        while (!cfManager.isMounted() && retryCount < 10)
-        {
-            std::this_thread::sleep_for(std::chrono::seconds(1));
-            retryCount++;
-        }
-
-        if (retryCount == 10)
+    // Mount might not show up immediately, so wait for mount to appear
+    int maxRetries = 10;
```

Comment:
Since you are doing a refactor, maybe we could remove this magic number and set a static int at the top of the file for this.

Reply:
That makes sense.
I needed to add a wait for unmount so I used it there too.